crates/graphql/src/schema.rs

```diff
@@ -166,12 +164,16 @@ impl BatchFn<String, Option<NftDetail>> for NftDetailBatcher {
 #[derive(Clone)]
 pub struct AppContext {
     nft_detail_loader: Loader<String, Option<NftDetail>, NftDetailBatcher>,
+    db_pool: Arc<Pool>,
 }
 
 impl AppContext {
-    pub fn new(db_connection: PooledConnection) -> AppContext {
+    pub fn new(db_pool: Arc<Pool>) -> AppContext {
         Self {
-            nft_detail_loader: Loader::new(NftDetailBatcher { db_connection }),
+            nft_detail_loader: Loader::new(NftDetailBatcher {
+                db_pool: db_pool.clone(),
+            }),
+            db_pool,
```

Comment:
Since arc is used clone each reference to db_pool besides the last within the block.